apps/aws-notebook/samples/ecs/fargate/vpn/main.tf

```diff
@@ -0,0 +1,252 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+}
+
+data "external" "tls" {
+  program = ["./prepare_openvpn_cert.sh"]
+}
+
+data "external" "vpn_client" {
+  program = ["./build_openvpn_config.sh"]
+
+  query = {
+    vpn_server_dns = aws_ec2_client_vpn_endpoint.dev-connection-ecs-fargate-vpn.dns_name
+  }
+}
+
+data "aws_iam_policy" "policy" {
+  name = "AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "role" {
+  name = "craftingDemoEcsTaskExecutionRole"
+
+  assume_role_policy = <<EOF
+    {
+      "Version": "2012-10-17",
+        "Statement": [
+        {
+          "Sid": "",
+          "Effect": "Allow",
+          "Principal": {
+            "Service": "ecs-tasks.amazonaws.com"
+          },
+          "Action": "sts:AssumeRole"
+        }
+        ]
+    }
+  EOF
+}
+
+resource "aws_iam_role_policy_attachment" "test-attach" {
+  role       = aws_iam_role.role.name
+  policy_arn = data.aws_iam_policy.policy.arn
+}
+
+data "aws_iam_role" "ecs_task_execution_role" {
+  #name = "ecsTaskExecutionRole"
+  name = aws_iam_role.role.name
+}
+
+resource "aws_security_group" "sg" {
+  name   = "allow_all"
+  vpc_id = aws_vpc.dev-connection-ecs-fargate-vpn.id
+  ingress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+    self             = true
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+    self             = true
+  }
+
+}
+
+# Create ECS cluster hosted on Fargate
+resource "aws_ecs_cluster" "dev-connection-ecs-fargate-vpn" {
+  name = "crafting-notebook-demo"
+
+  setting {
+    name  = "containerInsights"
+    value = "disabled"
+  }
+}
+
+# Create a VPC for ECS cluster
+resource "aws_vpc" "dev-connection-ecs-fargate-vpn" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+  tags = {
+    Name = "crafting-notebook-demo"
+  }
+}
+
+# Create a subnet for VPC
+resource "aws_subnet" "dev-connection-ecs-fargate-vpn" {
+  vpc_id     = aws_vpc.dev-connection-ecs-fargate-vpn.id
+  cidr_block = "10.0.0.0/17"
+
+  tags = {
+    Name = "crafting-notebook-demo-subnet-a"
+  }
```

Comment:
Please use default tags from the provider without specifying name repeatedly. All these can be tagged with same name or something like `demo-id` which is even better. Example: https://github.com/crafting-dev/sandbox-deploy/blob/master/tf/modules/sandbox/aws/vpc/main.tf#L11